numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -553,6 +557,13 @@ PyArray_OrderConverter(PyObject *object, NPY_ORDER *val)
     }
     else {
         str = PyBytes_AS_STRING(object);
+        if (strlen(str) != 1) {
+            DEPRECATE("Non length-one string passed in " \
```

Comment:
If you are curious: https://docs.python.org/2/c-api/exceptions.html#c.PyErr_WarnEx the second paragraph mentions it, the error handling code you can probably steal/infer from